src/Microsoft.DotNet.Tools.Builder/CompileContext.cs

```diff
@@ -52,22 +55,85 @@ public bool Compile(bool incremental)
             CreateOutputDirectories();
 
             //compile dependencies
-            foreach (var dependency in Sort(_dependencies))
+            foreach (var dependency in Sort(_dependencies.ProjectDependenciesWithSources))
             {
-                if (!InvokeCompileOnDependency(dependency, _outputPath, _intermediateOutputPath))
+                if (incremental)
+                {
+                    var dependencyContext = ProjectContext.Create(dependency.Path, dependency.TargetFrameworkInfo.FrameworkName);
```

Comment:
`dependency.TargetFrameworkInfo.FrameworkName` isn't correct. It should be using `dependency.Framework`